src/nvim/message.c

```diff
@@ -2478,8 +2471,10 @@ static void redir_write(char_u *str, int maxlen)
       }
     }
 
-    if (redir_reg) {
-      size_t len = maxlen == -1 ? STRLEN(s) : (size_t)maxlen;
+    size_t len = maxlen == -1 ? STRLEN(s) : (size_t)maxlen;
+    if (capture_ga) {
+      ga_concat_len(capture_ga, (const char *)str, len);
+    } else if (redir_reg) {
```

Comment:
@jamessan Not sure if I'm looking at it the wrong way, but I added a test that fails here:

https://github.com/justinmk/neovim/commit/abb3acfaa29aa68e24c0aa5febb5ffa9de6a24ee#diff-f6eb16769d15ab6e7a565f3a3b654332R72